src/utils.js

```diff
@@ -9,10 +11,41 @@ const isValidBoundingClientRect = clientRect => {
   return top != null && left != null && right != null && bottom != null;
 }
 
+/**
+* Returns whether or not the passed in object is a DOMRect, or at least is in
+* the shape of a DOMRect.
+*
+* @param rect: Object
+* @returns boolean
+*/
 export const isDOMRect = rect => {
   if (!window.DOMRect) {
-    return isValidBoundingClientRect(rect)
+    return isValidBoundingClientRect(rect);
+  }
+
+  return rect instanceof window.DOMRect;
+}
+
+/**
+* Returns an object representing the boundingClientRect for the given element,
+* or an empty object if not found.
+*
+* @param element: HTMLElement|ReactComponent
+* @returns Object
+*/
+export const getBoundingClientRectForElement = element => {
+  if (element != null) {
+    // native elements
+    if (element.getBoundingClientRect) {
+      return element.getBoundingClientRect();
+    }
+
+    // Assume React Element
+    const domNode = ReactDOM.findDOMNode(element)
+    if (domNode.getBoundingClientRect) {
+      return domNode.getBoundingClientRect();
+    }
   }
 
-  return rect instanceof window.DOMRect
+  return {};
 }
```

Comment:
Do you think it'd be better to return an empty object here or just throw an error? It seems like if we get to line 50 of this function throwing a descriptive error would be more helpful to developers.

Reply:
A great question! The only reason I had returned an empty object here is because we're relying on our new `isDomRect` function that's also here in `utils.js`, and you've got some pretty specific error messages up in popover.

from a purity standpoint, I could see why we would want the new util function `getBoundingClientRectForElement` to throw an error since we didn't get an actual DOMRect. Does this also mean we need to alter `Popover` so we get your nice messages instead of throwing out a generic error from the utility?

Reply:
I don't think so. I think we have enough information when we get to that point in the code that we can safely assume that the element they're passing in is not a React or HTMLElement.

I think something like:

```javascript
throw new ReferenceError("Either the prop \"parent\" or \"element\" is neither an instance of HTMLElement nor a React component")
```

What do you think about that? Also, I added build instructions to the bottom of the readme in case you want to run the webpack builds yourself.

Reply:
@brenthosie

Reply:
@RobertMenke 

I played around with this a little bit locally and I've got a couple of thoughts I want to run by you.

1) When I got to write the unit tests for the `utils` file, my error string is now talking about popover `parent` and `element` props instead of being a black box that accepts a reference to a native HTMLElement or React component and throws error strings talking about that reference being null. I think it would be good to have the error there be more generic.
2) We could go into `Popover`'s `willReceiveProps` function and before calling `setState` we could call out to the util file for both `parent` and `element`, surrounding their calls in try/catch and build a more helpful error message. We could say that either `parent` or `element` is the problem, or specify that they're both a problem at the same time.

if you think that's overkill, I could make the change you initially suggested and add your error message down in `utils`.

Reply:
@brenthosie 

1. I'm fine with the error being more generic. My main objective is for the error to be clear if the props supplied are incorrect.

2. I think that just 1 general error for both the `parent` and `element` props are fine. At that point, the developer should know that they need to check the props they're passing in.